extensions/wikia/Track/Track.php

```diff
@@ -1,59 +1,131 @@
+			   'x=' . $wgDBname . $sep .
+			   'y=' . $wgDBcluster . $sep .
+			   'u=' . $wgUser->getID() . $sep .
+			   'a=' . ( is_object( $wgArticle ) ? $wgArticle->getID() : null ) . $sep .
+			   's=' . RequestContext::getMain()->getSkin()->getSkinName() .
+			   ( $wgTitle && !is_object( $wgArticle ) ? $sep . 'pg=' . urlencode( $wgTitle->getPrefixedDBkey() ) : '' ) .
+			   ( $wgTitle ? $sep . 'n=' . $wgTitle->getNamespace() : '' ) .
+			   ( !empty( $wgAdServerTest ) ? $sep . 'db_test=1' : '' );
 
 		// Handle any parameters passed to us
-		if ($param) {
-			foreach ($param as $key => $val) {
-				$url .= $sep.urlencode($key).'='.urlencode($val);
+		if ( $param ) {
+			foreach ( $param as $key => $val ) {
+				$url .= $sep . urlencode( $key ) . '=' . urlencode( $val );
 			}
 		}
 
 		return $url;
 	}
 
-	private static function getViewJS ($param=null) {
+	private static function getGAURL( $type, $category, $action, $label, $value, $tid, $extraParams = [ ], $for_html = null ) {
+		global $wgTitle, $wgContLanguageCode, $wgDBname, $wgUser, $wgCityId, $wgGAUserIdSalt;
+
+		$skinName = RequestContext::getMain()->getSkin()->getSkinName();
+		$adContext = ( new AdEngine2ContextService() )->getContext( $wgTitle, $skinName );
+		$hubFactory = WikiFactoryHub::getInstance();
+
+		$params = [
+			'v' => static::GA_VERSION,
+			'ds' => 'backend',
+			't' => $type,
+			'ni' => 1,
+			'dl' => $wgTitle->getFullURL(),
+			'ul' => $wgContLanguageCode,
+			'de' => 'UTF-8',
+			'dt' => $wgTitle->getText(),
+			'ec' => $category,
+			'ea' => $action,
+			'el' => $label,
+			'ev' => $value,
+			'_utma' => $_COOKIE[ '__utma' ] ?? '',
+			'_utmz' => $_COOKIE[ '__utmz' ] ?? '',
+			'cid' => $_COOKIE[ '_ga' ] ? explode( ".", $_COOKIE[ '_ga' ] )[ 2 ] : uniqid(),
+			'tid' => $tid,
+			// custom dimensions
+			'cd1' => $wgDBname,
+			'cd2' => $wgContLanguageCode,
+			'cd3' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd4' => $skinName,
+			'cd5' => $wgUser->isAnon() ? 'anon' : 'user',
+			'cd8' => WikiaPageType::getPageType(),
+			'cd9' => $wgCityId,
+			'cd13' => AdTargeting::getEsrbRating(),
+			'cd14' => isset( $adContext[ 'opts' ][ 'showAds' ] ) ? 'Yes' : 'No',
+			'cd15' => WikiaPageType::isCorporatePage(),
+			'cd17' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd18' => implode( ',', $hubFactory->getWikiCategories( $wgCityId ) ),
+			'cd19' => WikiaPageType::getArticleType(),
+			'cd21' => $wgTitle->getArticleID(),
+			'cd23' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_FREQUENT ) ? 'Yes' : 'No',
+			'cd24' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_LIFETIME ) ? 'Yes' : 'No',
+			'cd25' => $wgTitle->getNamespace(),
+		];
+		if ( !$wgUser->isAnon() ) {
+			$params[ 'uid' ] = md5( $wgUser->getId() . $wgGAUserIdSalt );
+		}
+		$params = array_merge( $params, $extraParams );
+		$url = static::GA_URL . '/collect?' .
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$url". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)